src/test/java/net/snowflake/client/jdbc/PreparedStatementIT.java

```diff
@@ -0,0 +1,1721 @@
+  private final String deleteSQL = "delete from TEST_PREPST where ID = ?";
+  private final String selectSQL = "select * from TEST_PREPST where ID = ?";
+  private final String createTableSQL = "create or replace table test_prepst(id INTEGER, "
+                                        + "colA DOUBLE, colB FLOAT, colC String,  "
+                                        + "colD NUMBER, col INTEGER)";
+  private final String deleteTableSQL = "drop table if exists TEST_PREPST";
+  private final String enableCacheReuse = "alter session set USE_CACHED_RESULT=true";
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+
+  @Before
+  public void setUp() throws SQLException
+  {
+    Connection con = getConnection();
+    Statement st = con.createStatement();
+    st.execute(createTableSQL);
+    con.close();
+  }
+
+  @After
+  public void tearDown() throws SQLException
+  {
+    Connection con = getConnection();
+    Statement st = con.createStatement();
+    st.execute(deleteTableSQL);
+    con.close();
+  }
+
+  @Test
+  @ConditionalIgnore(condition = RunningOnTravisCI.class)
+  public void testInsertBatch() throws SQLException
+  {
+    int[] countResult;
+    connection = getConnection();
+    connection.createStatement().execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 0"); // disable stage bind
+    prepStatement = connection.prepareStatement(insertSQL);
+    bindOneParamSet(prepStatement, 1, 1.22222, (float) 1.2, "test", 12121212121L, (short) 12);
+    prepStatement.addBatch();
+    bindOneParamSet(prepStatement, 2, 2.22222, (float) 2.2, "test2", 1221221123131L, (short) 1);
+    prepStatement.addBatch();
+    countResult = prepStatement.executeBatch();
+    assertEquals(1, countResult[0]);
+    assertEquals(1, countResult[1]);
+    assertEquals(2, prepStatement.getUpdateCount());
+    resultSet = connection.createStatement().executeQuery(selectAllSQL);
+    assertEquals(2, getSizeOfResultSet(resultSet));
+    connection.close();
+  }
+
+  @Test
+  @ConditionalIgnore(condition = RunningOnTravisCI.class)
+  public void testInsertBatchStage() throws SQLException
+  {
+    int[] countResult;
+    connection = getConnection();
+    connection.createStatement().execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 12"); // enable stage bind
+    prepStatement = connection.prepareStatement(insertSQL);
+    bindOneParamSet(prepStatement, 1, 1.22222, (float) 1.2, "test", 12121212121L, (short) 12);
+    prepStatement.addBatch();
+    bindOneParamSet(prepStatement, 2, 2.22222, (float) 2.2, "test2", 1221221123131L, (short) 1);
+    prepStatement.addBatch();
+    countResult = prepStatement.executeBatch();
+    assertEquals(1, countResult[0]);
+    assertEquals(1, countResult[1]);
+    resultSet = connection.createStatement().executeQuery(selectAllSQL);
+    assertEquals(2, getSizeOfResultSet(resultSet));
+    connection.close();
+  }
```

Comment:
Can you add a large array binding test cases?
Since we cannot use the internal parameter CLIENT_STAGE_ARRAY_BINDIN_THRESHOLD, the tests should actually binding large data more than 65280, which is the current threshold. But the size is not too large to run it against the production.